src/content/developers/docs/scaling/optimistic-rollups/index.md

```diff
@@ -1,46 +1,254 @@
+
+The asserter is required to provide a “one-step proof” verifying the validity of the disputed single-step computation. If the asserter fails to provide the one-step proof, or the L1 verifier deems the proof invalid, they lose the challenge. 
+
+Some notes about this type of fraud proof: 
+
+1. Multi-round interactive fraud proving is considered efficient because it minimizes the work the L1 chain must do in dispute arbitration. Instead of replaying the entire transaction, Ethereum only needs to re-execute one step in the rollup's execution. 
+
+2. Bisection protocols reduce the amount of data posted on-chain (no need to publish state commits for every transaction). Also, optimistic rollup transactions don't have constrained by Ethereum's gas limit. Conversely, optimistic rollups re-executing transactions must make sure an L2 transaction has a lower gas limit to emulate its execution within a single Ethereum transaction.  
+
+##### Why fraud proofs matter for optimistic rollups {#fraud-proof-benefits}
+
+Fraud proofs are important because they facilitate *trustless finality* in optimistic rollups. Trustless finality is a quality of optimistic rollups that guarantees that a transaction—so long as it’s valid—will eventually be confirmed. 
+
+Malicious nodes can try to delay the confirmation of a valid rollup block by starting false challenges. However, fraud proofs will eventually prove the rollup block’s validity and cause it to be confirmed. 
+
+This also relates to another security property of optimistic rollups: the validity of the chain relies on the existence of *one* honest node. The honest node can advance the chain correctly by either posting valid assertions or disputing invalid assertions. Whatever the case, malicious nodes who enter into disputes with the honest node will lose their stakes during the fraud proving process. 
+
+### L1/L2 interoperability {#l1-l2-interoperability} 
+
+Optimistic rollups are designed for interoperability with Ethereum Mainnet and allow users to pass messages and arbitrary data between L1 and L2. They are also compatible with the EVM, so you can port existing [dapps](/developers/docs/dapps/) to rollups or create new dapps using Ethereum development tools. 
+
+#### 1. Asset movement {#asset-movement} 
+
+##### Entering the rollup {#entering-the-rollup} 
+
+To use an optimistic rollup, users deposit ETH, ERC-20 tokens, and other accepted assets in the rollup’s [bridge](/developers/docs/bridges/) contract on L1. The bridge contract will relay the transaction to L2, where an equivalent amount of assets is minted and sendt to the user’s chosen address on the optimistic rollup. 
+
+Typically, user-generated transactions (like an L1 > L2 deposit) are queued until the sequencer re-submits them to the rollup contract. However, to preserve censorship resistance, users are allowed to submit a transaction to the rollup contract if it has been delayed past the maximum time allowed. 
```

Comment:
Not sure I totally understand this part. I got this from Paradigm's [article](https://research.paradigm.xyz/optimism#data-availability-batches) on Optimism's rollup:

> Transactions are submitted to the CTC either by the sequencer, a prioritized party who can insert transactions into the chain, or via a first-in-first-out queue which feeds into the CTC. To preserve L1’s censorship resistance guarantees, anybody can submit transactions to this queue, forcing them to be included into the CTC after a delay.

And this is from the [Arbitrum docs](https://developer.offchainlabs.com/docs/inside_arbitrum#inboxes-fast-and-slow):

>Anyone else can submit a message, but messages submitted by non-sequencer nodes will be put into the "delayed inbox" queue, which is managed by an L1 Ethereum contract. Messages in the delayed inbox queue will wait there until the sequencer chooses to "release" them into the main inbox, where they will be added to the end of the inbox. A well-behaved sequencer will typically release delayed messages after about ten minutes, for reasons explained below. Alternatively, if a message has been in the delayed inbox queue for longer than a maximum delay interval (currently 24 hours) then anyone can force it to be promoted into the main inbox. (This ensures that the sequencer can only delay messages but can't censor them.)

Perhaps you can clarify what part of the text needs to change?

Reply:
I submitted a question about this to our protocol developers. I'll answer you when they answer me.

Reply:
Alright, then. LMK when you get a reply.

Reply:
OK, I got the reply.

What happens is bedrock (our new version, to be released in the second half of 2022) is that a block is defined as all the transactions submitted on L1 since the previous block (a.k.a. deposits) followed by all the transactions published by the sequencer. So if the sequencer ignores an L1 transaction it publishes the wrong (provably wrong) state root.

Reply:
So the sequencer can't even delay messages once they are posted on L1.

Reply:
> OK, I got the reply.
> 
> What happens is bedrock (our new version, to be released in the second half of 2022) is that a block is defined as all the transactions submitted on L1 since the previous block (a.k.a. deposits) followed by all the transactions published by the sequencer. So if the sequencer ignores an L1 transaction it publishes the wrong (provably wrong) state root.

So I've had some time to think about the page a bit, and I think it'd be ineffective to use Optimism's design as the sole basis of the content--especially as users will likely interact with other ORs. For instance, the censorship-resistance mechanism described here is the standard for Arbitrum, Boba, and Metis (the latter two are running forks of Optimism's current version).

Nevertheless, I also used your comment to add a new description of the censorship-resistance mechanism: 
> Some optimistic rollups adopt a more straightforward approach to prevent sequencers from censoring users. Here, a block is defined by all transactions submitted to the L1 contract since the previous block (e.g., deposits) in addition to the transactions processed on the rollup chain. If a sequencer ignores an L1 transaction, it will publish the (provably) wrong state root; therefore, sequencers cannot delay user-generated messages once posted on L1

Reply:
Great.